src/collections/components/AddLabelModal/AddLabelModal.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react';
+import { Grid } from '@mui/material';
+import { Modal } from '../../../_shared/components';
+import { useNotifications } from '../../../_shared/hooks';
+import { AddLabelForm } from '../';
+import { FormikHelpers, FormikValues } from 'formik';
+import { Label, useCreateLabelMutation } from '../../../api/generatedTypes';
+
+interface AddLabelModalProps {
+  /**
+   * Whether the modal is visible on the screen or not.
+   */
+  isOpen: boolean;
+
+  /**
+   * Toggle the AddLabelModalProps to show/hide as necessary.
+   */
+  toggleModal: VoidFunction;
+
+  /**
+   * state variable from Parent component LabelList Page
+   * */
+  setLabelsList: React.Dispatch<React.SetStateAction<Label[]>>;
+}
+
+/**
+ * Parent component for the AddProspectFormConnector component
+ */
+export const AddLabelModal: React.FC<AddLabelModalProps> = (
+  props
+): JSX.Element => {
+  const { isOpen, toggleModal, setLabelsList } = props;
+  const [isLoaderShowing, setIsLoaderShowing] = useState<boolean>(false);
+  const [createLabelMutation] = useCreateLabelMutation();
+  // Prepare state vars and helper methods for API notifications
+  const { showNotification } = useNotifications();
+
+  const onSubmit = (
+    values: FormikValues,
```

Comment:
There's quite a lot happening in this component. It's good that the API call is not being made straight from the Labels page as some of our older code does, but the other `XyzModal` components are not set up in the same way. They're solely responsible for Modal markup and managing their own state (e.g., open/closed). 

Then each either calls the form directly (older pattern) or call a Connector component (current pattern) which has next to no markup - it just calls the form and passes on the `onSubmit` function - but encapsulates all the business logic of calling the API and getting the response. Let's follow this pattern here. Then it will be relatively easy to set up unit tests for the UI-only components and integration tests for the component responsible for calling the API. Have a look at the flow from `ScheduleItemModal`, it's probably the most complete example.